### PR TITLE
Call evidence-api to reject a submitted document

### PR DESCRIPTION
**Pull request by bogdan-zaharia**
- Add the ability to reject submitted documents by calling the evidence-api and updating the state of the document submission as well as saving the rejection reason

### CODE REVIEW

File: src/components/RejectDialog.tsx [hunk 1]
```diff
@@ -4,13 +4,21 @@ import { Formik, Form } from 'formik';
...
-  reason: Yup.string()
+  rejectionReason: Yup.string()
     .required('Please give a reason')
     .min(2, 'Reason needs to be at least three characters'),
 });
 
+const initialValues = {
+  state: DocumentState.REJECTED,
+  rejectionReason: '',
+  staffSelectedDocumentTypeId: '',
```
Comment (neilmendum) [inline src/components/RejectDialog.tsx:19]: `staffSelectedDocumentTypeId` is optional on `DocumentSubmissionRequest` so I don't think we need to provide a default value here

**Review by neilmendum — COMMENTED** [review]
Looks good! Made one small comment

File: src/components/RejectDialog.tsx (hunk 1 above)
Comment (bogdan-zaharia) [reply src/components/RejectDialog.tsx]: Good point, addressed

**Review by neilmendum — APPROVED** [review]
LGTM

**Pull request merged**